Src/LibHoney/Transmission.cs

```diff
@@ -23,34 +27,48 @@ class Transmission : IDisposable
 
         const string HoneyEventsUrl = "/1/events/";
 
-        public Transmission (bool blockOnSend, bool blockOnResponse)
+        public Transmission (int maxConcurrentBatches, bool blockOnSend, bool blockOnResponse)
         {
             BlockOnSend = blockOnSend;
             BlockOnResponse = blockOnResponse;
 
             pending = new BlockingCollection<Event> (MaxPendingEvents);
+            responses = new BlockingCollection<Response> (MaxPendingResponses);
 
             client = new HttpClient ();
             client.DefaultRequestHeaders.UserAgent.ParseAdd (HoneyUserAgent);
             client.Timeout = TimeSpan.FromSeconds (TimeoutInSeconds);
 
-            InitBackgroundSendTask ();
+            this.maxConcurrentBatches = maxConcurrentBatches;
+            InitBackgroundSendThreads ();
         }
 
-        // Used a single thread, from which we do all the delivering,
-        // as we dont want to pollute the threadpool. See:
-        // https://channel9.msdn.com/Events/TechEd/Europe/2013/DEV-B318
-        void InitBackgroundSendTask ()
+        // Use old nice Thread objects, as we do not want to touch
+        // the threadpool, and Task objects make use of it.
+        // Also catch all the exceptions, as not handling them would
+        // crash the *entire* application to crash.
+        void InitBackgroundSendThreads ()
         {
-            sendTask = Task.Run (() => {
-                while (true) {
-                    var ev = pending.Take ();
-                    if (ev == null) // Signaled we are stopping the task.
-                        return;
-
-                    DoSend (ev);
-                }
-            });
+            countdownEv = new CountdownEvent (maxConcurrentBatches);
+
+            for (int i = 0; i < maxConcurrentBatches; i++) {
+                var t = new Thread (() => {
+                    try {
+                        var ev = pending.Take ();
+                        if (ev == null) // Signaled we are stopping the task.
+                            return;
+
+                        DoSend (ev);
+                    } catch (Exception) {
+                        // (xxx) calberto: Where should we log this error?
```

Comment:
hm, the more I think about it, `DoSend` shouldn't let any exceptions escape - handling them in `DoSend` and enqueuing the correct response sounds like the right thing to do, right?